src/backend/distributed/commands/utility_hook.c

```diff
@@ -891,14 +891,24 @@ ShouldCheckUndistributeCitusLocalTables(void)
 static bool
 ShouldAddNewTableToMetadata(Node *parsetree)
 {
-	if (!IsA(parsetree, CreateStmt))
+	CreateStmt *createTableStmt;
```

Comment:
It seems `CREATE FOREIGN TABLE .. PARTITION OF` doesn't honor `use_citus_managed_tables` GUC. For example, in the following example, `issues_0/1/2` does not become citus local tables, which should have been.

```SQL
SET citus.use_citus_managed_tables TO ON;

CREATE TABLE public.issues (
    project_id integer
) PARTITION BY HASH (project_id);


CREATE SERVER IF NOT EXISTS part1 FOREIGN DATA WRAPPER postgres_fdw OPTIONS (dbname 'otherdb1',host 'localhost', port '5432');
CREATE SERVER IF NOT EXISTS part2 FOREIGN DATA WRAPPER postgres_fdw OPTIONS (dbname 'otherdb2', host 'localhost', port '5432');
CREATE SERVER IF NOT EXISTS part3 FOREIGN DATA WRAPPER postgres_fdw OPTIONS (dbname 'otherdb3', host 'localhost', port '5432');

CREATE FOREIGN TABLE issues_0 PARTITION OF public.issues FOR VALUES WITH (modulus 3, remainder 0) SERVER part1;
CREATE FOREIGN TABLE issues_1 PARTITION OF public.issues FOR VALUES WITH (modulus 3, remainder 1) SERVER part2;
CREATE FOREIGN TABLE issues_2 PARTITION OF public.issues FOR VALUES WITH (modulus 3, remainder 2) SERVER part3;
 select * from pg_dist_partition ;
 logicalrelid | partmethod | partkey | colocationid | repmodel | autoconverted 
--------------+------------+---------+--------------+----------+---------------
 issues       | n          |         |            0 | s        | f
(1 row)


```

In general, make sure to add a partitioned foreign table test.

Reply:
another set of steps that should work:

```SQL
CREATE TABLE public.issues (
    project_id integer
) PARTITION BY HASH (project_id);


CREATE SERVER IF NOT EXISTS part1 FOREIGN DATA WRAPPER postgres_fdw OPTIONS (dbname 'otherdb1',host 'localhost', port '5432');
CREATE SERVER IF NOT EXISTS part2 FOREIGN DATA WRAPPER postgres_fdw OPTIONS (dbname 'otherdb2', host 'localhost', port '5432');
CREATE SERVER IF NOT EXISTS part3 FOREIGN DATA WRAPPER postgres_fdw OPTIONS (dbname 'otherdb3', host 'localhost', port '5432');

CREATE FOREIGN TABLE issues_0 PARTITION OF public.issues FOR VALUES WITH (modulus 3, remainder 0) SERVER part1;
CREATE FOREIGN TABLE issues_1 PARTITION OF public.issues FOR VALUES WITH (modulus 3, remainder 1) SERVER part2;
CREATE FOREIGN TABLE issues_2 PARTITION OF public.issues FOR VALUES WITH (modulus 3, remainder 2) SERVER part3;


SELECT citus_add_local_table_to_metadata('issues');
NOTICE:  extension "postgres_fdw" already exists, skipping
NOTICE:  server "part1" already exists, skipping
NOTICE:  extension "postgres_fdw" already exists, skipping
NOTICE:  server "part2" already exists, skipping
NOTICE:  extension "postgres_fdw" already exists, skipping
NOTICE:  server "part3" already exists, skipping
ERROR:  non-citus partitioned tables cannot have citus table partitions
HINT:  Distribute the partitioned table "issues_102015" instead, or add it to metadata
Time: 275.382 ms
```

Reply:
fixed, thanks.
Pushed with tests

Reply:
Would be great to add the second set of the tests where we explicitly call `citus_add_local_table_to_metadata` instead of relaying on the GUC